### PR TITLE
Revert "Allow additive-expr with operands of a union type containing builtin subtypes of string and xml"

### PR DESCRIPTION
**Pull request by MaryamZi**
Reverts ballerina-platform/ballerina-lang#34817

Since it was to the 2201.0.x branch.

### CODE REVIEW

**Review by gimantha — APPROVED** [review]

**Review by gimantha — APPROVED** [review]

File: language-server/modules/langserver-core/src/test/resources/completion/statement_context/config/onfail_clause_ctx_config2c.json [hunk 1]
```diff
@@ -254,6 +254,102 @@
       "filterText": "var",
       "insertText": "var ",
       "insertTextFormat": "Snippet"
+    },
```
Comment (MaryamZi) [inline language-server/modules/langserver-core/src/test/resources/completion/statement_context/config/onfail_clause_ctx_config2c.json:257]: @gimantha, why do we need this change in this PR? It just reverts the previous lang PR (https://github.com/ballerina-platform/ballerina-lang/pull/34817)? Was it already failing on 2201.0.x?

Comment (gimantha) [reply language-server/modules/langserver-core/src/test/resources/completion/statement_context/config/onfail_clause_ctx_config2c.json:257]: Yes. it failed. https://github.com/ballerina-platform/ballerina-lang/runs/5890888206?check_suite_focus=true

Comment (MaryamZi) [reply language-server/modules/langserver-core/src/test/resources/completion/statement_context/config/onfail_clause_ctx_config2c.json:257]: Thanks. 

Seems to have been failing in https://github.com/ballerina-platform/ballerina-lang/runs/5880386248?check_suite_focus=true also. cc @mohanvive @malinthar

**Review by malinthar — APPROVED** [review]
LGTM!

**Comment by MaryamZi** [comment]
@keizer619, @anupama-pathirage, @HindujaB, can we get this merged? Reverts a PR we merged to 2201.0.x by mistake and also fixes a lang server failure on the branch.

**Pull request merged**